### PR TITLE
Unify logic for laying out floats and atomic inlines

### PR DESCRIPTION
**Pull request by Loirooriol**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because there is no behavior change

### CODE REVIEW

File: components/layout_2020/flow/inline/mod.rs [hunk 1]
```diff
@@ -2039,58 +2028,85 @@ impl IndependentFormattingContext {
...
+            return fragment;
+        }
+
+        // Offset the content rectangle by the physical offset of the padding, border, and margin.
+        let pbm_physical_offset = pbm_sums
+            .start_offset()
+            .to_physical_size(container_writing_mode);
+        fragment.content_rect = fragment
+            .content_rect
+            .translate(pbm_physical_offset.to_vector());
```
Comment (mrobinson) [inline components/layout_2020/flow/inline/mod.rs:2071]: This feels like it belongs a bit closer to inline layout, could you return `pbm_sums` from this function and have it take place in inline layout?

File: components/layout_2020/flow/inline/mod.rs [hunk 2]
```diff
@@ -2039,58 +2028,85 @@ impl IndependentFormattingContext {
...
+
         // Apply baselines if necessary.
-        let mut fragment = match baselines {
+        match baselines {
             Some(baselines) => fragment.with_baselines(baselines),
             None => fragment,
-        };
+        }
```
Comment (mrobinson) [inline components/layout_2020/flow/inline/mod.rs:2077]: You could also return `baselines` and inline layout take care of that. Perhaps just return a data structure called `IndependentFormattingContextLayout`?

**Review by mrobinson — APPROVED** [review]
Looks good. The only other thing than what I've mentioned below is that this should probably move to a different file such as `flow/mod.rs` or `flow/block.rs`. Not sure.

File: components/layout_2020/flow/inline/mod.rs (hunk 2 above)
Comment (Loirooriol) [reply components/layout_2020/flow/inline/mod.rs]: Went with `IndependentLayoutResult`, since it's similar to `IndependentLayout`, but the result of that.

**Comment by Loirooriol** [comment]
Moved the code to `flow/mod.rs` (there is no `flow/block.rs`)

**Pull request merged**